snuba/query/processors/single_tag_expr.py

```diff
@@ -0,0 +1,90 @@
+from typing import Mapping, Iterable
+
+from snuba.clickhouse.columns import ColumnSet
+from snuba.query.expressions import (
+    Column,
+    Expression,
+    FunctionCall,
+    Literal,
+    NestedColumn,
+)
+from snuba.query.dsl import array_element
+from snuba.query.query import Query
+from snuba.query.query_processor import QueryProcessor
+from snuba.request.request_settings import RequestSettings
+
+
+class SingleTagProcessor(QueryProcessor):
```

Comment:
I decided not to reuse the same code we use for today's tag parsing (like I did for conditions and functions a few months ago) since here the code is simpler, the AST version is substantially different and I had the opportunity to clean it up a bit.

Again, see the unit test for an example of this code.